Throw concrete exception when no accessToken is provided

When not providing an accessToken through `Mapbox.java`, 
an Activity containing our MapView will crash with:

```
com.mapbox.mapboxsdk.plugins.testapp / com.mapbox.mapboxsdk.plugins.testapp.TrafficActivity
}: android.view.InflateException: Binary XML file line #10: Error inflating class com.mapbox.mapboxsdk.maps.MapView
                                                                                          at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2358)
                                                                                          at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2420)
                                                                                          at android.app.ActivityThread.access$900(ActivityThread.java:154)
                                                                                          at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1321)
                                                                                          at android.os.Handler.dispatchMessage(Handler.java:102)
                                                                                          at android.os.Looper.loop(Looper.java:135)
                                                                                          at android.app.ActivityThread.main(ActivityThread.java:5292)
                                                                                          at java.lang.reflect.Method.invoke(Native Method)
                                                                                          at java.lang.reflect.Method.invoke(Method.java:372)
                                                                                          at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:904)
                                                                                          at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:699)
                                                                                       Caused by: android.view.InflateException: Binary XML file line # 10: Error inflating class com.mapbox.mapboxsdk.maps.MapView
at android.view.LayoutInflater.createView(LayoutInflater.java: 633)
at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java: 743)
at android.view.LayoutInflater.rInflate(LayoutInflater.java: 806)
at android.view.LayoutInflater.inflate(LayoutInflater.java: 504)
at android.view.LayoutInflater.inflate(LayoutInflater.java: 414)
at android.view.LayoutInflater.inflate(LayoutInflater.java: 365)
at android.support.v7.app.AppCompatDelegateImplV9.setContentView(AppCompatDelegateImplV9.java: 292)
at android.support.v7.app.AppCompatActivity.setContentView(AppCompatActivity.java: 140)
at com.mapbox.mapboxsdk.plugins.testapp.TrafficActivity.onCreate(TrafficActivity.java: 25)
at android.app.Activity.performCreate(Activity.java: 5990)
at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java: 1106)
at android.app.ActivityThread.performLaunchActivity(ActivityThread.java: 2311)
at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java: 2420) 
at android.app.ActivityThread.access$900(ActivityThread.java: 154) 
at android.app.ActivityThread$H.handleMessage(ActivityThread.java: 1321) 
at android.os.Handler.dispatchMessage(Handler.java: 102) 
at android.os.Looper.loop(Looper.java: 135) 
at android.app.ActivityThread.main(ActivityThread.java: 5292) 
at java.lang.reflect.Method.invoke(Native Method) 
at java.lang.reflect.Method.invoke(Method.java: 372) 
at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java: 904) 
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java: 699) 
Caused by: java.lang.reflect.InvocationTargetException
at java.lang.reflect.Constructor.newInstance(Native Method)
at java.lang.reflect.Constructor.newInstance(Constructor.java: 288)
at android.view.LayoutInflater.createView(LayoutInflater.java: 607)
at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java: 743) 
at android.view.LayoutInflater.rInflate(LayoutInflater.java: 806) 
at android.view.LayoutInflater.inflate(LayoutInflater.java: 504) 
at android.view.LayoutInflater.inflate(LayoutInflater.java: 414) 
at android.view.LayoutInflater.inflate(LayoutInflater.java: 365) 
at android.support.v7.app.AppCompatDelegateImplV9.setContentView(AppCompatDelegateImplV9.java: 292) 
at android.support.v7.app.AppCompatActivity.setContentView(AppCompatActivity.java: 140) 
at com.mapbox.mapboxsdk.plugins.testapp.TrafficActivity.onCreate(TrafficActivity.java: 25) 
at android.app.Activity.performCreate(Activity.java: 5990) 
at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java: 1106) 
at android.app.ActivityThread.performLaunchActivity(ActivityThread.java: 2311) 
at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java: 2420) 
at android.app.ActivityThread.access$900(ActivityThread.java: 154) 
at android.app.ActivityThread$H.handleMessage(ActivityThread.java: 1321) 
at android.os.Handler.dispatchMessage(Handler.java: 102) 
at android.os.Looper.loop(Looper.java: 135) 
at android.app.ActivityThread.main(ActivityThread.java: 5292) 
at java.lang.reflect.Method.invoke(Native Method) 
at java.lang.reflect.Method.invoke(Method.java: 372) 
at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java: 904) 
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java: 699) 
Caused by: java.lang.NullPointerException: Attempt to read from field 'java.lang.String com.mapbox.mapboxsdk.Mapbox.accessToken'
on a null object reference
at com.mapbox.mapboxsdk.Mapbox.validateAccessToken(Mapbox.java: 68)
at com.mapbox.mapboxsdk.Mapbox.getAccessToken(Mapbox.java: 58)
at com.mapbox.mapboxsdk.storage.FileSource. < init > (FileSource.java: 111)
at com.mapbox.mapboxsdk.storage.FileSource.getInstance(FileSource.java: 44)
at com.mapbox.mapboxsdk.maps.NativeMapView. < init > (NativeMapView.java: 78)
at com.mapbox.mapboxsdk.maps.MapView.initialise(MapView.java: 126)
at com.mapbox.mapboxsdk.maps.MapView. < init > (MapView.java: 93)
at java.lang.reflect.Constructor.newInstance(Native Method) 
at java.lang.reflect.Constructor.newInstance(Constructor.java: 288) 
at android.view.LayoutInflater.createView(LayoutInflater.java: 607) 
at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java: 743) 
at android.view.LayoutInflater.rInflate(LayoutInflater.java: 806) 
at android.view.LayoutInflater.inflate(LayoutInflater.java: 504) 
at android.view.LayoutInflater.inflate(LayoutInflater.java: 414) 
at android.view.LayoutInflater.inflate(LayoutInflater.java: 365) 
at android.support.v7.app.AppCompatDelegateImplV9.setContentView(AppCompatDelegateImplV9.java: 292) 
at android.support.v7.app.AppCompatActivity.setContentView(AppCompatActivity.java: 140) 
at com.mapbox.mapboxsdk.plugins.testapp.TrafficActivity.onCreate(TrafficActivity.java: 25) 
at android.app.Activity.performCreate(Activity.java: 5990) 
at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java: 1106) 
at android.app.ActivityThread.performLaunchActivity(ActivityThread.java: 2311) 
at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java: 2420) 
at android.app.ActivityThread.access$900(ActivityThread.java: 154) 
at android.app.ActivityThread$H.handleMessage(ActivityThread.java: 1321) 
at android.os.Handler.dispatchMessage(Handler.java: 102) 
at android.os.Looper.loop(Looper.java: 135) 
at android.app.ActivityThread.main(ActivityThread.java: 5292) 
at java.lang.reflect.Method.invoke(Native Method) 
at java.lang.reflect.Method.invoke(Method.java: 372) 
at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java: 904) 
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java: 699) 
```

This is a regression as we use to show an InvalidAccessTokenException to the developer instead.

--- a/platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/Mapbox.java
+++ b/platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/Mapbox.java
@@ -7,7 +7,7 @@ import android.support.annotation.NonNull;
 import android.text.TextUtils;
 
 import com.mapbox.mapboxsdk.constants.MapboxConstants;
-import com.mapbox.mapboxsdk.exceptions.InvalidAccessTokenException;
+import com.mapbox.mapboxsdk.exceptions.MapboxConfigurationException;
 import com.mapbox.mapboxsdk.location.LocationSource;
 import com.mapbox.mapboxsdk.net.ConnectivityReceiver;
 import com.mapbox.services.android.telemetry.MapboxTelemetry;
@@ -55,20 +55,30 @@ public final class Mapbox {
    * @return Mapbox Access Token
    */
   public static String getAccessToken() {
+    validateMapbox();
     validateAccessToken();
     return INSTANCE.accessToken;
   }
 
   /**
+   * Runtime validation of Mapbox creation.
+   */
+  private static void validateMapbox() throws MapboxConfigurationException {
+    if (INSTANCE == null) {
+      throw new MapboxConfigurationException();
+    }
+  }
+
+  /**
    * Runtime validation of Access Token.
    *
-   * @throws InvalidAccessTokenException exception thrown when not using a valid accessToken
+   * @throws MapboxConfigurationException exception thrown when not using a valid accessToken
    */
-  private static void validateAccessToken() throws InvalidAccessTokenException {
+  private static void validateAccessToken() throws MapboxConfigurationException {
     String accessToken = INSTANCE.accessToken;
     if (TextUtils.isEmpty(accessToken) || (!accessToken.toLowerCase(MapboxConstants.MAPBOX_LOCALE).startsWith("pk.")
       && !accessToken.toLowerCase(MapboxConstants.MAPBOX_LOCALE).startsWith("sk."))) {
-      throw new InvalidAccessTokenException();
+      throw new MapboxConfigurationException();
     }
   }
 

--- a/platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/exceptions/MapboxConfigurationException.java
+++ b/platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/exceptions/MapboxConfigurationException.java
@@ -1,21 +1,18 @@
 package com.mapbox.mapboxsdk.exceptions;
 
 import android.content.Context;
-import android.os.Bundle;
-
-import com.mapbox.mapboxsdk.maps.MapView;
 
 /**
- * A {@code InvalidAccessTokenException} is thrown by {@link com.mapbox.mapboxsdk.maps.MapboxMap}
- * when there is either no access token set before {@link MapView#onCreate(Bundle)} or an invalid access token
- * is set in {@link com.mapbox.mapboxsdk.Mapbox#getInstance(Context, String)}
- *
- * @see MapView#onCreate(Bundle)
+ * A MapboxConfigurationException is thrown by MapboxMap when the SDK hasn't been properly initialised.
+ * <p>
+ * This occurs either when {@link com.mapbox.mapboxsdk.Mapbox} is not correctly initialised or the provided access token
+ * through {@link com.mapbox.mapboxsdk.Mapbox#getInstance(Context, String)} isn't valid.
+ * </p>
  * @see com.mapbox.mapboxsdk.Mapbox#getInstance(Context, String)
  */
-public class InvalidAccessTokenException extends RuntimeException {
+public class MapboxConfigurationException extends RuntimeException {
 
-  public InvalidAccessTokenException() {
+  public MapboxConfigurationException() {
     super("\nUsing MapView requires setting a valid access token. Use Mapbox.getInstance(Context context, "
       + "String accessToken) to provide one. "
       + "\nPlease see https://www.mapbox.com/help/create-api-access-token/ to learn how to create one."

--- a/platform/android/MapboxGLAndroidSDKTestApp/src/test/java/com/mapbox/mapboxsdk/MapboxTest.java
+++ b/platform/android/MapboxGLAndroidSDKTestApp/src/test/java/com/mapbox/mapboxsdk/MapboxTest.java
@@ -4,7 +4,7 @@ import android.content.Context;
 import android.net.ConnectivityManager;
 import android.net.NetworkInfo;
 
-import com.mapbox.mapboxsdk.exceptions.InvalidAccessTokenException;
+import com.mapbox.mapboxsdk.exceptions.MapboxConfigurationException;
 
 import org.junit.Before;
 import org.junit.Test;
@@ -39,7 +39,7 @@ public class MapboxTest {
     assertSame(accessToken, Mapbox.getAccessToken());
   }
 
-  @Test(expected = InvalidAccessTokenException.class)
+  @Test(expected = MapboxConfigurationException.class)
   public void testGetInvalidAccessToken() {
     final String accessToken = "dummy";
     injectMapboxSingleton(accessToken);

